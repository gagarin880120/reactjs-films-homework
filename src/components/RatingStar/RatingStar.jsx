import React from 'react';
import PropTypes from 'prop-types';

export default function RatingStar({ type }) {
  return (
    <>
      {
        type === 'full'
          ? <i className="fas fa-star" />
          : <i className="fas fa-star-half" />
      }
    </>
  );
}

RatingStar.propTypes = {
  type: PropTypes.string,
}

RatingStar.defaultProps = {
  type: '',
}
