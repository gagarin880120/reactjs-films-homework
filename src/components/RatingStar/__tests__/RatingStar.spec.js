import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import RatingStar from '../RatingStar';

describe('RatingStar component', () => {
  test('should render full star icon when type is "full"', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<RatingStar type="full" />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  test('should render half star icon when type is "half"', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<RatingStar type="half" />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
