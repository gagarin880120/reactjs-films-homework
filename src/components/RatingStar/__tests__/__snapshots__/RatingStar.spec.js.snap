// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingStar component should render full star icon when type is "full" 1`] = `
<i
  className="fas fa-star"
/>
`;

exports[`RatingStar component should render half star icon when type is "half" 1`] = `
<i
  className="fas fa-star-half"
/>
`;
